Python 3.10.0b1: test failures (due to DeprecationWarnings) + hang

Werkzeug's tests fail with Python 3.10.0b1. Furthermore, the test suite seems to hang at the end, I had to interrupt with with `^c`.

```pytb
$ tox -e py310
[...]
==================================================================== FAILURES ====================================================================
____________________________________________________________ test_basic_local_asyncio ____________________________________________________________
tests/test_local.py:58: in test_basic_local_asyncio
    loop = asyncio.get_event_loop()
E   DeprecationWarning: There is no current event loop
____________________________________________________________ test_local_stack_asyncio ____________________________________________________________
tests/test_local.py:121: in test_local_stack_asyncio
    loop = asyncio.get_event_loop()
E   DeprecationWarning: There is no current event loop
________________________________________________________________ test_proxy_await ________________________________________________________________
tests/test_local.py:574: in test_proxy_await
    out = asyncio.get_event_loop().run_until_complete(main())
E   DeprecationWarning: There is no current event loop
________________________________________________________________ test_proxy_aiter ________________________________________________________________
tests/test_local.py:602: in test_proxy_aiter
    out = asyncio.get_event_loop().run_until_complete(main())
E   DeprecationWarning: There is no current event loop
________________________________________________________ test_proxy_async_context_manager ________________________________________________________
tests/test_local.py:626: in test_proxy_async_context_manager
    assert asyncio.get_event_loop().run_until_complete(main())
E   DeprecationWarning: There is no current event loop
_______________________________________________________________ test_server[https] _______________________________________________________________
tests/test_serving.py:42: in test_server
    client = dev_server(**kwargs)
tests/conftest.py:129: in start_dev_server
    _, log_path = xprocess.ensure(xp_name, Starter, restart=True)
.tox/py310/lib/python3.10/site-packages/xprocess.py:168: in ensure
    if not starter.wait(self.file_handles[-1]):
.tox/py310/lib/python3.10/site-packages/xprocess.py:261: in wait
    has_match = any(std.re.search(self.pattern, line) for line in lines)
.tox/py310/lib/python3.10/site-packages/xprocess.py:261: in <genexpr>
    has_match = any(std.re.search(self.pattern, line) for line in lines)
.tox/py310/lib/python3.10/site-packages/werkzeug/utils.py:97: in __get__
    value = self.fget(obj)  # type: ignore
tests/conftest.py:124: in pattern
    client.request("/ensure")
tests/conftest.py:81: in request
    conn = self.connect()
tests/conftest.py:69: in connect
    kwargs["context"] = ssl.SSLContext()
/usr/lib/python3.10/ssl.py:494: in __new__
    warnings.warn(
E   DeprecationWarning: ssl module: SSLContext() without protocol argument is deprecated.
-------------------------------------------------------------- Captured stdout call --------------------------------------------------------------
/tmp/werkzeug/.tox/py310/.pytest_cache/d/.xprocess/dev_server-test_server[https]$ /tmp/werkzeug/.tox/py310/bin/python /tmp/werkzeug/tests/live_apps/run.py standard {"ssl_context": "adhoc", "port": 49153}
process 'dev_server-test_server[https]' started pid=2552350
 * Running on https://127.0.0.1:49153/ (Press CTRL+C to quit)

_______________________________________________________________ test_ssl_dev_cert ________________________________________________________________
tests/test_serving.py:77: in test_ssl_dev_cert
    client = dev_server(ssl_context=make_ssl_devcert(tmp_path))
tests/conftest.py:129: in start_dev_server
    _, log_path = xprocess.ensure(xp_name, Starter, restart=True)
.tox/py310/lib/python3.10/site-packages/xprocess.py:168: in ensure
    if not starter.wait(self.file_handles[-1]):
.tox/py310/lib/python3.10/site-packages/xprocess.py:261: in wait
    has_match = any(std.re.search(self.pattern, line) for line in lines)
.tox/py310/lib/python3.10/site-packages/xprocess.py:261: in <genexpr>
    has_match = any(std.re.search(self.pattern, line) for line in lines)
.tox/py310/lib/python3.10/site-packages/werkzeug/utils.py:97: in __get__
    value = self.fget(obj)  # type: ignore
tests/conftest.py:124: in pattern
    client.request("/ensure")
tests/conftest.py:81: in request
    conn = self.connect()
tests/conftest.py:69: in connect
    kwargs["context"] = ssl.SSLContext()
/usr/lib/python3.10/ssl.py:494: in __new__
    warnings.warn(
E   DeprecationWarning: ssl module: SSLContext() without protocol argument is deprecated.
-------------------------------------------------------------- Captured stdout call --------------------------------------------------------------
/tmp/werkzeug/.tox/py310/.pytest_cache/d/.xprocess/dev_server-test_ssl_dev_cert$ /tmp/werkzeug/.tox/py310/bin/python /tmp/werkzeug/tests/live_apps/run.py standard {"ssl_context": ["/tmp/werkzeug/.tox/py310/tmp/test_ssl_dev_cert0.crt", "/tmp/werkzeug/.tox/py310/tmp/test_ssl_dev_cert0.key"], "port": 49154}
process 'dev_server-test_ssl_dev_cert' started pid=2552367
 * Running on https://127.0.0.1:49154/ (Press CTRL+C to quit)

________________________________________________________________ test_ssl_object _________________________________________________________________
tests/test_serving.py:84: in test_ssl_object
    client = dev_server(ssl_context="custom")
tests/conftest.py:129: in start_dev_server
    _, log_path = xprocess.ensure(xp_name, Starter, restart=True)
.tox/py310/lib/python3.10/site-packages/xprocess.py:168: in ensure
    if not starter.wait(self.file_handles[-1]):
.tox/py310/lib/python3.10/site-packages/xprocess.py:261: in wait
    has_match = any(std.re.search(self.pattern, line) for line in lines)
.tox/py310/lib/python3.10/site-packages/xprocess.py:261: in <genexpr>
    has_match = any(std.re.search(self.pattern, line) for line in lines)
.tox/py310/lib/python3.10/site-packages/werkzeug/utils.py:97: in __get__
    value = self.fget(obj)  # type: ignore
tests/conftest.py:124: in pattern
    client.request("/ensure")
tests/conftest.py:81: in request
    conn = self.connect()
tests/conftest.py:69: in connect
    kwargs["context"] = ssl.SSLContext()
/usr/lib/python3.10/ssl.py:494: in __new__
    warnings.warn(
E   DeprecationWarning: ssl module: SSLContext() without protocol argument is deprecated.
-------------------------------------------------------------- Captured stdout call --------------------------------------------------------------
/tmp/werkzeug/.tox/py310/.pytest_cache/d/.xprocess/dev_server-test_ssl_object$ /tmp/werkzeug/.tox/py310/bin/python /tmp/werkzeug/tests/live_apps/run.py standard {"ssl_context": "custom", "port": 49155}
process 'dev_server-test_ssl_object' started pid=2552368
 * Running on https://127.0.0.1:49155/ (Press CTRL+C to quit)

============================================================ short test summary info =============================================================
FAILED tests/test_local.py::test_basic_local_asyncio - DeprecationWarning: There is no current event loop
FAILED tests/test_local.py::test_local_stack_asyncio - DeprecationWarning: There is no current event loop
FAILED tests/test_local.py::test_proxy_await - DeprecationWarning: There is no current event loop
FAILED tests/test_local.py::test_proxy_aiter - DeprecationWarning: There is no current event loop
FAILED tests/test_local.py::test_proxy_async_context_manager - DeprecationWarning: There is no current event loop
FAILED tests/test_serving.py::test_server[https] - DeprecationWarning: ssl module: SSLContext() without protocol argument is deprecated.
FAILED tests/test_serving.py::test_ssl_dev_cert - DeprecationWarning: ssl module: SSLContext() without protocol argument is deprecated.
FAILED tests/test_serving.py::test_ssl_object - DeprecationWarning: ssl module: SSLContext() without protocol argument is deprecated.
=================================================== 8 failed, 792 passed, 1 skipped in 42.19s ====================================================
Traceback (most recent call last):
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/_pytest/main.py", line 269, in wrap_session
    session.exitstatus = doit(config, session) or 0
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/_pytest/main.py", line 323, in _main
    config.hook.pytest_runtestloop(session=session)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/callers.py", line 208, in _multicall
    return outcome.get_result()
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/_pytest/main.py", line 348, in pytest_runtestloop
    item.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/callers.py", line 208, in _multicall
    return outcome.get_result()
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/_pytest/runner.py", line 109, in pytest_runtest_protocol
    runtestprotocol(item, nextitem=nextitem)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/_pytest/runner.py", line 127, in runtestprotocol
    reports.append(call_and_report(item, "teardown", log, nextitem=nextitem))
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/_pytest/runner.py", line 215, in call_and_report
    call = call_runtest_hook(item, when, **kwds)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/_pytest/runner.py", line 254, in call_runtest_hook
    return CallInfo.from_call(
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/_pytest/runner.py", line 311, in from_call
    result: Optional[TResult] = func()
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/_pytest/runner.py", line 255, in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/callers.py", line 208, in _multicall
    return outcome.get_result()
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/_pytest/runner.py", line 175, in pytest_runtest_teardown
    item.session._setupstate.teardown_exact(item, nextitem)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/_pytest/runner.py", line 419, in teardown_exact
    self._teardown_towards(needed_collectors)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/_pytest/runner.py", line 427, in _teardown_towards
    self._pop_and_teardown()
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/_pytest/runner.py", line 387, in _pop_and_teardown
    self._teardown_with_finalization(colitem)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/_pytest/runner.py", line 405, in _teardown_with_finalization
    self._callfinalizers(colitem)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/_pytest/runner.py", line 395, in _callfinalizers
    fin()
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/_pytest/fixtures.py", line 1034, in finish
    raise exc
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/_pytest/fixtures.py", line 1027, in finish
    func()
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/_pytest/fixtures.py", line 941, in _teardown_yield_fixture
    next(it)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pytest_xprocess.py", line 48, in xprocess
    p.wait(xproc.proc_wait_timeout)
  File "/usr/lib/python3.10/subprocess.py", line 1203, in wait
    return self._wait(timeout=timeout)
  File "/usr/lib/python3.10/subprocess.py", line 1931, in _wait
    time.sleep(delay)
KeyboardInterrupt

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/werkzeug/.tox/py310/bin/pytest", line 8, in <module>
    sys.exit(console_main())
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/_pytest/config/__init__.py", line 185, in console_main
    code = main()
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/_pytest/config/__init__.py", line 162, in main
    ret: Union[ExitCode, int] = config.hook.pytest_cmdline_main(
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/callers.py", line 208, in _multicall
    return outcome.get_result()
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/_pytest/main.py", line 316, in pytest_cmdline_main
    return wrap_session(config, _main)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/_pytest/main.py", line 283, in wrap_session
    config.hook.pytest_keyboard_interrupt(excinfo=excinfo)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/callers.py", line 208, in _multicall
    return outcome.get_result()
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/_pytest/terminal.py", line 837, in pytest_keyboard_interrupt
    self._keyboardinterrupt_memo = excinfo.getrepr(funcargs=True)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/_pytest/_code/code.py", line 648, in getrepr
    return fmt.repr_excinfo(self)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/_pytest/_code/code.py", line 905, in repr_excinfo
    reprtraceback = self.repr_traceback(excinfo_)
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/_pytest/_code/code.py", line 828, in repr_traceback
    traceback = excinfo.traceback
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/_pytest/_code/code.py", line 545, in traceback
    self._traceback = Traceback(self.tb, excinfo=ref(self))
  File "/tmp/werkzeug/.tox/py310/lib/python3.10/site-packages/_pytest/_code/code.py", line 333, in __init__
    super().__init__(f(tb))
KeyboardInterrupt
/usr/lib/python3.10/subprocess.py:1066: ResourceWarning: subprocess 2552350 is still running
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='/tmp/werkzeug/.tox/py310/.pytest_cache/d/.xprocess/dev_server-test_ssl_object/xprocess.log' mode='r' encoding='UTF-8'>
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='/tmp/werkzeug/.tox/py310/.pytest_cache/d/.xprocess/dev_server-test_ssl_dev_cert/xprocess.log' mode='r' encoding='UTF-8'>
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='/tmp/werkzeug/.tox/py310/.pytest_cache/d/.xprocess/dev_server-test_server[https]/xprocess.log' mode='r' encoding='UTF-8'>
/usr/lib/python3.10/subprocess.py:1066: ResourceWarning: subprocess 2552410 is still running
/usr/lib/python3.10/subprocess.py:1066: ResourceWarning: subprocess 2552409 is still running
/usr/lib/python3.10/subprocess.py:1066: ResourceWarning: subprocess 2552408 is still running
/usr/lib/python3.10/subprocess.py:1066: ResourceWarning: subprocess 2552407 is still running
/usr/lib/python3.10/subprocess.py:1066: ResourceWarning: subprocess 2552406 is still running
/usr/lib/python3.10/subprocess.py:1066: ResourceWarning: subprocess 2552404 is still running
/usr/lib/python3.10/subprocess.py:1066: ResourceWarning: subprocess 2552403 is still running
/usr/lib/python3.10/subprocess.py:1066: ResourceWarning: subprocess 2552380 is still running
/usr/lib/python3.10/subprocess.py:1066: ResourceWarning: subprocess 2552371 is still running
/usr/lib/python3.10/subprocess.py:1066: ResourceWarning: subprocess 2552368 is still running
/usr/lib/python3.10/subprocess.py:1066: ResourceWarning: subprocess 2552367 is still running
/usr/lib/python3.10/subprocess.py:1066: ResourceWarning: subprocess 2552366 is still running
/usr/lib/python3.10/subprocess.py:1066: ResourceWarning: subprocess 2552365 is still running
/usr/lib/python3.10/subprocess.py:1066: ResourceWarning: subprocess 2552364 is still running
/usr/lib/python3.10/subprocess.py:1066: ResourceWarning: subprocess 2552363 is still running
/usr/lib/python3.10/subprocess.py:1066: ResourceWarning: subprocess 2552351 is still running
____________________________________________________________________ summary _____________________________________________________________________
ERROR:   py310: keyboardinterrupt
```

Environment:

- Python version: 3.10.0b1
- Werkzeug version: b01fa1817343d2a36a9d8bb17f61ddf209c27c2b

Not sure I understand what you mean. Do you want me to repaste this?

```
============================================================ short test summary info =============================================================
FAILED tests/test_local.py::test_basic_local_asyncio - DeprecationWarning: There is no current event loop
FAILED tests/test_local.py::test_local_stack_asyncio - DeprecationWarning: There is no current event loop
FAILED tests/test_local.py::test_proxy_await - DeprecationWarning: There is no current event loop
FAILED tests/test_local.py::test_proxy_aiter - DeprecationWarning: There is no current event loop
FAILED tests/test_local.py::test_proxy_async_context_manager - DeprecationWarning: There is no current event loop
FAILED tests/test_serving.py::test_server[https] - DeprecationWarning: ssl module: SSLContext() without protocol argument is deprecated.
FAILED tests/test_serving.py::test_ssl_dev_cert - DeprecationWarning: ssl module: SSLContext() without protocol argument is deprecated.
FAILED tests/test_serving.py::test_ssl_object - DeprecationWarning: ssl module: SSLContext() without protocol argument is deprecated.
```

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -66,7 +66,10 @@ class DevServerClient:
 
         if protocol == "https":
             if "context" not in kwargs:
-                kwargs["context"] = ssl.SSLContext()
+                context = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
+                context.check_hostname = False
+                context.verify_mode = ssl.CERT_NONE
+                kwargs["context"] = context
 
             return http.client.HTTPSConnection(self.addr, **kwargs)
 

--- a/tests/test_local.py
+++ b/tests/test_local.py
@@ -12,6 +12,18 @@ import pytest
 from werkzeug import local
 
 
+if sys.version_info < (3, 7):
+
+    def run_async(coro):
+        return asyncio.get_event_loop().run_until_complete(coro)
+
+
+else:
+
+    def run_async(coro):
+        return asyncio.run(coro)
+
+
 def test_basic_local():
     ns = local.Local()
     ns.foo = 0
@@ -55,9 +67,11 @@ def test_basic_local_asyncio():
         await asyncio.sleep(0.02)
         values.append(ns.foo)
 
-    loop = asyncio.get_event_loop()
-    futures = [asyncio.ensure_future(value_setter(idx)) for idx in [1, 2, 3]]
-    loop.run_until_complete(asyncio.gather(*futures))
+    async def main():
+        futures = [asyncio.ensure_future(value_setter(i)) for i in [1, 2, 3]]
+        await asyncio.gather(*futures)
+
+    run_async(main())
     assert sorted(values) == [1, 2, 3]
 
     def delfoo():
@@ -118,9 +132,11 @@ def test_local_stack_asyncio():
         ls.push(1)
         assert len(ls._local.stack) == 2
 
-    loop = asyncio.get_event_loop()
-    futures = [asyncio.ensure_future(task()) for _ in range(3)]
-    loop.run_until_complete(asyncio.gather(*futures))
+    async def main():
+        futures = [asyncio.ensure_future(task()) for _ in range(3)]
+        await asyncio.gather(*futures)
+
+    run_async(main())
 
 
 @pytest.mark.skipif(
@@ -571,7 +587,7 @@ def test_proxy_await():
     async def main():
         return await p
 
-    out = asyncio.get_event_loop().run_until_complete(main())
+    out = run_async(main())
     assert out == 1
 
 
@@ -599,7 +615,7 @@ def test_proxy_aiter():
 
         return out
 
-    out = asyncio.get_event_loop().run_until_complete(main())
+    out = run_async(main())
     assert out == [2, 1, 0]
 
 
@@ -623,4 +639,4 @@ def test_proxy_async_context_manager():
         assert p.value == 2
         return True
 
-    assert asyncio.get_event_loop().run_until_complete(main())
+    assert run_async(main())

